Objects/listobject.c

```diff
@@ -2659,38 +2659,56 @@ list.index
     start: slice_index(accept={int}) = 0
     stop: slice_index(accept={int}, c_default="PY_SSIZE_T_MAX") = sys.maxsize
     /
+    *
+    key: object=NULL
 
 Return first index of value.
 
 Raises ValueError if the value is not present.
+A callable 'key' can be provided to transform each item before comparison with 'value'.
 [clinic start generated code]*/
 
 static PyObject *
 list_index_impl(PyListObject *self, PyObject *value, Py_ssize_t start,
-                Py_ssize_t stop)
-/*[clinic end generated code: output=ec51b88787e4e481 input=40ec5826303a0eb1]*/
+                Py_ssize_t stop, PyObject *key)
+/*[clinic end generated code: output=23a18266d9a3a9b3 input=7115b8d01b704c54]*/
 {
     Py_ssize_t i;
 
     if (start < 0) {
         start += Py_SIZE(self);
-        if (start < 0)
+        if (start < 0) {
```

Comment:
I've seen other comments, but I have an oposite opinion: diff should be as minimal as possible. This is just a comment, no action needed, unless others also agree :)

Reply:
I disagree, the style should be consistent within the function.  Either I tick to pep7 in the entire function, or the old style.